Assets/Scripts/Managers/BoardManager.cs

```diff
@@ -330,6 +331,8 @@ public ChessPiece CreatePiece(ChessPieceType type, (int, int) pos, Team team)
         piece.gameObject.transform.position = square.CenterSurfaceTransform.position;
         piece.Team = team;
         piece.AssignedSquare = square;
+        piece.OnSuccessfulUpgrade += _audioManager.Value.PlaySuccessSFX;
+        piece.OnFailedUpgrade += _audioManager.Value.PlayFailSFX;
```

Comment:
We also want to remove subscriptions when we don't need them anymore (Can cause issues).

Also easy to do. we unsubscribe by `-=` instead of `+=`

So in the OnDestroy method. 

```
piece.OnSuccessfulUpgrade -= _audioManager.Value.PlaySuccessSFX;
piece.OnFailedUpgrade -= _audioManager.Value.PlayFailSFX;
```


In this case it might be easier to let the Pieces just directly fire off the `_audioManager` calls within the `ChessPiece.cs` code.
Then we dont have to have the events / subscriptions. or worry about cleaning them up.


Also remember you can get access to services ANYWHERE. so the chesspieces themselves can gain access to the audioManager